modules/oxide-components/src/test/ts/browser/keynav/TabbingType.spec.tsx

```
@@ -0,0 +1,290 @@
+
+    await userEvent.keyboard('{Escape}');
+    expect(store, 'Check that escape handler has fired').toEqual([ 'tab.onEscape' ]);
+
+  });
+
+  it('Should handle keyboard navigation using Tab key in a tree structure', async () => {
+
+    const TabKeyingInTree = () => {
+      const outerRef = useRef<HTMLDivElement>(null);
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      const innerRefs = useMemo(() => new Array(3).fill(null).map(() => useRef<HTMLDivElement>(null)), []);
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)